rpc/client.go

```diff
@@ -114,15 +127,19 @@ func (c *Client) writePump(logger log.TMLogger) {
 			}
 
 			if err := c.conn.WriteMessage(websocket.TextMessage, message); err != nil {
-				logger.Error("error writing message to websocket", "err", err)
+				if err.Error() != "io: read/write on closed pipe" {
```

Comment:
Err, nope, not gonna do that. Why does it matter what kind of error it is? Just log it regardless.

Reply:
`io: read/write on closed pipe` error occurs whenever the client closes the WebSocket connection. So it's not really an error so much as being informed that the dialogue is over.
Perhaps give an Info message, rather than an error, to record when the WebSocket connect closed?

Reply:
You can check for `err != io.ErrClosedPipe` then. It doesn't seem like logging any of these socket errors as `logger.Error` is particularly useful, maybe `logger.Debug` instead... in case anyone ever needs to debug something. Keep using `logger.Error` for the panics though, those shouldn't be happening often I presume.

Reply:
Two types of error.
1)  Errors that occur repeatedly during normal operation. (e.g. closed socket and EOF)
2)  Errors that normally never happen.
So report 1) as debug logs and 2) as error logs.